src/local/int_cvx.lean

```diff
@@ -0,0 +1,102 @@
+import data.real.basic
+import linear_algebra.basic
+import linear_algebra.finite_dimensional
+import category_theory.endomorphism
+import topology.algebra.module
+import topology.instances.real
+import analysis.convex.basic
+
+-- This is intended as a Lean roadmap for the `int_cvx` lemma, currently Lemma 1.5 of loops.tex.
+-- Rather than a "human readable" proof of a single lemma,
+-- it is a longer sequence of lemmas, facts, and constructions, for now all sorried.
+
+-- TODO everything in this file could be generalised to arbitrary affine spaces over ℝ.
+variables {V : Type*} [add_comm_group V] [vector_space ℝ V]
+
+/-- The dilation about `c` with scale factor `r`. -/
+def dilation (c : V) (r : ℝ) : V → V := sorry
+
+/-- The dilations about `c` give a multiplicative homomorphism from ℝ to End V. -/
+-- In fact, dilations preserve the affine structure,
+-- so one could strengthen this to `ℝ →* End (Aff.of V)`,
+-- if we define `Aff`.
+def dilations (c : V) : ℝ →* category_theory.End V := sorry
```

Comment:
Do you expect to be able to use the category library here, or is it only for completeness?

Reply:
I only did this because it's the only mechanism I know of to quickly put a monoid structure on endomorphisms. This is the same reason the category theory API gets imported on the way to the Hahn-Banach theorem in mathlib!